diagnosis/getkuberneteslogs.sh

```diff
@@ -290,20 +300,41 @@ fi
 
 echo "[$(date +%Y%m%d%H%M%S)][INFO] Done collecting Kubernetes logs"
 
-# TODO Move inside UPLOAD_LOGS
-echo "[$(date +%Y%m%d%H%M%S)][INFO] Processing logs"
-createSADirectories
-
-for log in $(ls ${LOGFILEFOLDER}/*/containers/*.log)
-do
-    CNAME=$(basename ${log} .log)
-    CMETA=${LOGFILEFOLDER}/cluster-snapshot-$NOW/${CNAME}.meta
+if [ -n "$UPLOAD_LOGS" ]; then
+    echo "[$(date +%Y%m%d%H%M%S)][INFO] Processing logs"
+    createSADirectories
     
-    CLOG=${SA_DIR}/${CNAME}.log
-    echo "== BEGIN HEADER ==" > ${CLOG}
-    jq -r 'to_entries|map("\(.key): \(.value|tostring)")|.[]' ${CMETA} >> ${CLOG}
-    echo "== END HEADER ==" >> ${CLOG}
-    cat ${log} >> ${CLOG}
-done
+    for log in $(ls ${LOGFILEFOLDER}/*/containers/*.log)
+    do
+        CNAME=$(basename ${log} .log)
+        CMETA=${LOGFILEFOLDER}/cluster-snapshot-$NOW/${CNAME}.meta
+        
+        CLOG=${SA_DIR}/${CNAME}.log
+        echo "== BEGIN HEADER ==" > ${CLOG}
+        jq -r 'to_entries|map("\(.key): \(.value|tostring)")|.[]' ${CMETA} >> ${CLOG}
+        echo "== END HEADER ==" >> ${CLOG}
+        cat ${log} >> ${CLOG}
+    done
+    
+    #storage account variables
+    SA_NAME="kubernetesdiagnostics"
+    SA_RESOURCE_GROUP="kubernetesdiagnostics"
+    
+    #create resource group for kubernetes disgnostics
+    if [ $(az group exists --name $SA_RESOURCE_GROUP) = false ]; then
+        az group create -n $SA_RESOURCE_GROUP -l $LOCATION
```

Comment:
could you please move lines 323-330 to a new function called `ensureResourceGroup` (or similar)